File: Python/013__Roman_To_Integer.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        map = {"I":1, "V":5, "X":10, "L":50, "C":100, "D":500, "M":1000}
        result = 0
        front = map[s[0]]
        for i in range(len(s) - 1):
            next = map[s[i + 1]]
            if front < next:
                result -= front
            else:
                result += front
            front = next
        result += map[s[-1]]

        return result

File: Java/013__Roman_to_Integer.java
class Solution {
    public int romanToInt(String s) {
        HashMap<Character, Integer> map = new HashMap<>();
        map.put('I', 1);
        map.put('V', 5);
        map.put('X', 10);
        map.put('L', 50);
        map.put('C', 100);
        map.put('D', 500);
        map.put('M', 1000);
        int result = 0;
        char[] s_arr = s.toCharArray();
        for (int i = 0; i < s_arr.length - 1; i++) {
            if (map.get(s_arr[i]) < map.get(s_arr[i + 1])) {
                result -= map.get(s_arr[i]);
            } else {
                result += map.get(s_arr[i]);
            }
        }
        result += map.get(s_arr[s_arr.length - 1]);
        return result;
    }
}
